doc/spec-files.rst

```diff
@@ -370,49 +370,44 @@ of ``Info.plist``.)
 
 |PyInstaller| creates a minimal :file:`Info.plist`.
 You can add or overwrite entries in the plist by passing an
-``info_plist=`` parameter to the BUNDLE call.
-The value of this argument is a Python dict.
-Each key and value in the dict becomes a key and value in the
-:file:`Info.plist` file.
-For example, when you use PyQt5,
-you can set ``NSHighResolutionCapable`` to ``True`` to let your app
-also work in retina screen::
+``info_plist=`` parameter to the BUNDLE call.  Its argument should be a
+Python dict with keys and values to be included in the :file:`Info.plist`
+file.
+|PyInstaller| creates :file:`Info.plist` from the info_plist dict
+using the Python Standard Library module plistlib_.
+plistlib can handle nested python objects (which are translated to nested
+XML.)  Python data types are translated to the proper :file:`Info.plist`
+XML types.  Here's an example::
 
     app = BUNDLE(exe,
              name='myscript.app',
              icon=None,
-             bundle_identifier=None
+             bundle_identifier=None,
              info_plist={
-             	'NSHighResolutionCapable': 'True'
+             	'NSHighResolutionCapable': True,
+                'CFBundleDocumentTypes': [
+                    {
+                        'CFBundleTypeName': 'My File Format',
+                        'CFBundleTypeIconFile': 'MyFileIcon.icns',
+                        'LSItemContentTypes': ['com.example.myformat'],
+                        'LSHandlerRank': 'Owner'
+                        }
+                    ]
              	},
              )
 
-The ``info_plist=`` parameter only handles simple key:value pairs.
-It cannot handle nested XML arrays.
-For example, if you want to modify :file:`Info.plist` to tell Mac OS X
-what filetypes your app supports, you must add a 
-``CFBundleDocumentTypes`` entry to :file:`Info.plist`
-(see `Apple document types`_).
-The value of that keyword is a list of dicts,
-each containing up to five key:value pairs.
-
-To add such a value to your app's :file:`Info.plist` you must edit the
-plist file separately after |PyInstaller| has created the app.
-However, when you re-run |PyInstaller|, your changes will be wiped out.
-One solution is to prepare a complete :file:`Info.plist` file and
-copy it into the app after creating it.
-
-Begin by building and testing the windowed app.
-When it works, copy the ``Info.plist`` prepared by |PyInstaller|.
-This includes the ``CFBundleExecutable`` value as well as the
-icon path and bundle identifier if you supplied them.
-Edit the ``Info.plist`` as necessary to add more items
-and save it separately.
-
-From that point on, to rebuild the app call |PyInstaller| in a shell script,
-and follow it with a statement such as::
-
-    cp -f Info.plist dist/myscript.app/Contents/Info.plist
+For example, when you use PyQt5,
```

Comment:
Is NSHighResolutionCapable related to PyQt5? If not, I'd remove the "when you use PyQt5,".

Reply:
This was existing in the documentation, and I didn't understand it either so I left it.  To tell you the truth I've never used NSHighResolutionCapable.  Usually the Info.plist key that's crucial for an application to have retina resolution enabled is `'NSPrincipleClass': 'NSApplication'`.

Reply:
https://developer.apple.com/library/content/documentation/General/Reference/InfoPlistKeyReference/Articles/CocoaKeys.html#//apple_ref/doc/uid/TP40009251-SW11
https://developer.apple.com/library/content/documentation/GraphicsAnimation/Conceptual/HighResolutionOSX/Explained/Explained.html

Based on those two links, I'm starting to think that NSHighResolutionCapable is a red herring and should be removed from the example.

It seems that the crucial thing to enable retina is to inform macOS that the application is a Cocoa app, using `'NSPrincipleClass': 'NSApplication'`.  Then, the default value of `NSHighResolutionCapable` should already be `True`